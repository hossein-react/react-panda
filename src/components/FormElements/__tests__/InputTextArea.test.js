import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import InputTextArea from '../InputTextArea';

global.Input = require('antd').Input;

describe('<InputTextArea />', () => {
  const tree = props =>
    renderer
      .create(
        <MemoryRouter>
          <InputTextArea
            {...props}
            input={{ name: 'field' }}
            meta={{ touched: false, error: null }}
          />
        </MemoryRouter>
      )
      .toJSON();

  it('renders  without error', () => {
    const props = {
      input: { name: 'field' },
      label: 'field',
      placeholder: 'placeholder',
      meta: { touched: false, error: null }
    };
    expect(tree(props)).toMatchSnapshot();
  });
  it('renders with values', () => {
    const props = {
      input: { name: 'field' },
      label: 'field',
      value: 'Some text',
      meta: { touched: true, error: null }
    };
    expect(tree(props)).toMatchSnapshot();
  });
  it('renders error, not touched', () => {
    const props = {
      input: { name: 'field' },
      label: 'field',
      meta: { touched: false, error: 'Required' }
    };
    expect(tree(props)).toMatchSnapshot();
  });

  it('renders error, and touched', () => {
    const props = {
      input: { name: 'field' },
      label: 'field',
      meta: { touched: true, error: 'Required' }
    };
    expect(tree(props)).toMatchSnapshot();
  });

  it('renders disabled', () => {
    const props = {
      input: { name: 'field' },
      label: 'field',
      meta: { touched: false, error: null },
      disabled: true
    };
    expect(tree(props)).toMatchSnapshot();
  });
});
