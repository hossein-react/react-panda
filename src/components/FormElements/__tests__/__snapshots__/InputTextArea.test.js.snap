// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputTextArea /> Test blur method 1`] = `
<div
  className="c-input textarea undefined "
>
  <label
    htmlFor="field"
  >
    <span
      className="labelText"
    >
      field
    </span>
    <textarea
      className="ant-input roundInput textareaInput"
      id="field"
      name="field"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="placeholder"
      rows={5}
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </label>
</div>
`;

exports[`<InputTextArea /> renders  without error 1`] = `
<div
  className="c-input textarea undefined "
>
  <label
    htmlFor="field"
  >
    <span
      className="labelText"
    >
      field
    </span>
    <textarea
      className="ant-input roundInput textareaInput"
      id="field"
      name="field"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="placeholder"
      rows={5}
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </label>
</div>
`;

exports[`<InputTextArea /> renders disabled 1`] = `
<div
  className="c-input textarea undefined "
>
  <label
    htmlFor="field"
  >
    <span
      className="labelText"
    >
      field
    </span>
    <textarea
      className="ant-input roundInput textareaInput ant-input-disabled"
      disabled={true}
      id="field"
      name="field"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      rows={5}
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </label>
</div>
`;

exports[`<InputTextArea /> renders error, and touched 1`] = `
<div
  className="c-input textarea undefined "
>
  <label
    htmlFor="field"
  >
    <span
      className="labelText"
    >
      field
    </span>
    <textarea
      className="ant-input roundInput textareaInput"
      id="field"
      name="field"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      rows={5}
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </label>
</div>
`;

exports[`<InputTextArea /> renders error, not touched 1`] = `
<div
  className="c-input textarea undefined "
>
  <label
    htmlFor="field"
  >
    <span
      className="labelText"
    >
      field
    </span>
    <textarea
      className="ant-input roundInput textareaInput"
      id="field"
      name="field"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      rows={5}
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </label>
</div>
`;

exports[`<InputTextArea /> renders with values 1`] = `
<div
  className="c-input textarea undefined "
>
  <label
    htmlFor="field"
  >
    <span
      className="labelText"
    >
      field
    </span>
    <textarea
      className="ant-input roundInput textareaInput"
      id="field"
      name="field"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      rows={5}
      style={
        Object {
          "width": "100%",
        }
      }
      value="Some text"
    />
  </label>
</div>
`;
